src/hotspot/share/prims/nativeEntryPoint.cpp

```diff
@@ -52,10 +52,12 @@ JNI_ENTRY(jlong, NEP_makeDowncallStub(JNIEnv* env, jclass _unused, jobject metho
   GrowableArray<VMStorage> input_regs(pcount);
   for (int i = 0, bt_idx = 0; i < pcount; i++) {
     oop type_oop = java_lang_invoke_MethodType::ptype(type, i);
-    assert(java_lang_Class::is_primitive(type_oop), "Only primitives expected");
-    BasicType bt = java_lang_Class::primitive_type(type_oop);
+    BasicType bt = java_lang_Class::as_BasicType(type_oop);
     basic_type[bt_idx++] = bt;
-    input_regs.push(ForeignGlobals::parse_vmstorage(arg_moves_oop->obj_at(i)));
+    oop reg_oop = arg_moves_oop->obj_at(i);
+    if (reg_oop != nullptr) {
+      input_regs.push(ForeignGlobals::parse_vmstorage(reg_oop));
+    }
```

Comment:
Some registers are `null`, which indicates that the corresponding value passed to the downcall stub should not be forwarded to the native call, but is instead used directly in the downcall stub (e.g. the offset of an oop).